consensus/src/main/java/org/apache/iotdb/consensus/multileader/MultiLeaderConsensus.java

```diff
@@ -310,13 +310,25 @@ public ConsensusGenericResponse removePeer(ConsensusGroupId groupId, Peer peer)
           .build();
     }
     try {
+      // let other peers remove the sync channel with target peer
       impl.notifyPeersToRemoveSyncLogChannel(peer);
     } catch (ConsensusGroupAddPeerException e) {
       return ConsensusGenericResponse.newBuilder()
           .setSuccess(false)
           .setException(new ConsensusException(e.getMessage()))
           .build();
     }
+
+    try {
+      // let target peer reject new write
+      impl.inactivePeer(peer);
+      // wait its SyncLog to complete
+      impl.waitTargetPeerUntilSyncLogCompleted(peer);
+    } catch (ConsensusGroupAddPeerException e) {
+      // we only log warning here because sometimes the target peer may already be down
+      logger.warn("cannot wait {} to complete SyncLog. error message: {}", peer, e.getMessage());
+    }
```

Comment:
Why we catch `ConsensusGroupAddPeerException` in removePeer method? I think we need to specific the exception

Reply:
Sure, I will define another similar exception